### ISSUE TITLE
Model property to indicate if the last saved version is valid or not (shex)

### ISSUE THREAD

**Issue by lpalbou** [geneontology/minerva#246]
The following proposal was discussed this morning with both @vanaukenk and @tmushayahama .

**Back end proposal**: when requesting list of models from NF or ART, add a property in the minerva response to indicate if each model is valid with respect to GO-CAM specs (shex).

**Front end proposal**: invalid models would be identified with an icon (e.g. :exclamation:). Invalid facts within the model would be highlighted for instance with a red border on NF fields.

**Tech proposal**: having a script that can be executed on a set of models (1+) and update each model property. The script would be run anytime a model is going to be saved and anytime the shex model changes (recompute on all models). @goodb @kltm @balhoff .

Priority for this is still to be decided.

**Comment by kltm** [comment]
It would be good to better understand the context that generated this proposal. Possibly something to talk about this week on the software call.

**Comment by lpalbou** [comment]
👍 

For a little context: @tmushayahama was concerned about NF/ART having to deal with invalid models and if he had to implement additional checks on the UI side. In the long terms, NF/ART can probably trust that all saved models will be valid but not for the moment. Having a tag/property stating if a model is valid would help his error handling on UI side. It would also benefit curators who would instantly see if a model is valid or if it requires rework.

At this stage, this is just a proposal discussed this morning and no priority has been set yet by @vanaukenk .

**Comment by tmushayahama** [comment]
yes, @lpalbou @kltm  this will be helpful. As explained above, having a model level annotation check of whether the model is valid or not (last saved). Just a boolean for now. Parsing a wrong model into activities and error handling without much knowledge of data is very difficult on the client  Things like carnality, There are actually 2 use cases here.

- First is NF for chunking (parsing into activities) and for ordinary error checking just like the current simple annoton editor (while waiting for the minerva validation).
- ART is to flag the models fast as the model level annotations can be Sparql'ed quickly with @goodb search API. (To recap, during search mode, ART displays just a list first and model level annotations)

**Comment by goodb** [comment]
Consensus from GO Berkeley, this is a good idea.  Lets look into it it further.

**Comment by balhoff** [comment]
Something to consider is when to re-check all models (which would require loading each into memory), for example since the ontology or ShEx schema may change after validation was performed.

**Comment by kltm** [comment]
Is the proposal to have a flag, or the contents of the last report?

**Comment by goodb** [comment]
I believe the idea is to add a parameter with values = valid, invalid, unknown
Not sure if we want this to cover both shex and owl check or if we want a separate check.  
Underlying reason is for identifying models that curators need to pay attention to in search result views.
